dm/dm/master/shardddl/optimist.go

```diff
@@ -117,9 +117,14 @@ func (o *Optimist) Locks() map[string]*optimism.Lock {
 }
 
 // ShowLocks is used by `show-ddl-locks` command.
-func (o *Optimist) ShowLocks(task string, sources []string) []*pb.DDLLock {
+func (o *Optimist) ShowLocks(task string, sources []string) ([]*pb.DDLLock, error) {
 	locks := o.lk.Locks()
 	ret := make([]*pb.DDLLock, 0, len(locks))
+	var ifm map[string]map[string]map[string]map[string]optimism.Info
+	opm, _, err := optimism.GetAllOperations(o.cli)
+	if err == nil {
+		ifm, _, err = optimism.GetAllInfo(o.cli)
+	}
```

Comment:
Can we return err here earlier?

Reply:
This error will affect users to check `owners` and `ddlGroups`. However, it doesn't affect them getting ddl locks' information. So I think maybe a warning is enough here.